metricflow-semantics/metricflow_semantics/specs/spec_classes.py

```diff
@@ -681,42 +681,55 @@ class JoinToTimeSpineDescription:
 
 @dataclass(frozen=True)
 class GroupByMetricSpec(LinkableInstanceSpec, SerializableDataclass):
-    """Metric used in group by or where filter."""
+    """Metric used in group by or where filter.
+
+    Args:
+        element_name: Name of the metric being joined.
+        entity_links: Sequence of entities joined to join the metric subquery to the outer query. Last entity is the one
+            joining the subquery to the outer query.
+        metric_subquery_entity_links: Sequence of entities used in the metric subquery to join the metric to the entity.
+            Does not include the top-level entity (to avoid duplicating the last element of `entity_links`).
+    """
+
+    metric_subquery_entity_links: Tuple[EntityReference, ...]
 
     @property
     def without_first_entity_link(self) -> GroupByMetricSpec:  # noqa: D102
         assert len(self.entity_links) > 0, f"Spec does not have any entity links: {self}"
-        return GroupByMetricSpec(element_name=self.element_name, entity_links=self.entity_links[1:])
+        return GroupByMetricSpec(
+            element_name=self.element_name,
+            entity_links=self.entity_links[1:],
+            metric_subquery_entity_links=self.metric_subquery_entity_links,
+        )
 
     @property
     def without_entity_links(self) -> GroupByMetricSpec:  # noqa: D102
-        return GroupByMetricSpec(element_name=self.element_name, entity_links=())
+        return GroupByMetricSpec(
+            element_name=self.element_name,
+            entity_links=(),
+            metric_subquery_entity_links=self.metric_subquery_entity_links,
+        )
 
     @property
     def last_entity_link(self) -> EntityReference:  # noqa: D102
         assert len(self.entity_links) > 0, f"Spec does not have any entity links: {self}"
         return self.entity_links[-1]
 
-    @staticmethod
```

Comment:
Not used, and seems less useful for this spec considering you can't pass the subquery entity links via string name.